app/components/article-featured-video.js

```diff
@@ -23,7 +25,35 @@ export default Component.extend(
 		didRender() {
 			this._super(...arguments);
 
-			this.initVideoPlayer();
+			if (!this.get('hasRendered')) {
+				this.initVideoPlayer();
+				this.initOnScrollBehaviour();
+				this.set('hasRendered', true);
+			}
+		},
+
+		/**
+		 * Manages video transformation on user's scroll action
+		 *
+		 * @returns {void}
+		 */
+		initOnScrollBehaviour() {
+			let $video = this.$('.video-container'),
+				videoBottomPosition = $video.offset().top + $video.height();
+
+			this.$(window).on('scroll.featured-video', () => {
+				run.throttle(this, () => {
+					let currentScroll = this.$(window).scrollTop();
+
+					if (currentScroll >= videoBottomPosition && this.canVideoDrawerShow()) {
+						this.set('isVideoDrawerVisible', true);
+						this.toggleSiteHeadShadow(false);
+					} else if (currentScroll < videoBottomPosition - $video.height()) {
```

Comment:
`$video.height()` checks the DOM state which is kinda expensive. Can we have `videoTopPosition = $video.offset().top` defined together with `videoBottomPosition` or is the height dynamic in this case?